mlflow/store/tracking/abstract_store.py

```diff
@@ -243,13 +243,42 @@ def set_tag(self, run_id, tag):
         """
         self.log_batch(run_id, metrics=[], params=[], tags=[tag])
 
+    def get_metric_history(self, run_id, metric_key, max_results=None, page_token=None):
+        """
+        Return a list of metric objects corresponding to all values logged for a given metric
+        within a run.
+
+        See ``_get_metric_history`` for parameter descriptions.
+
+        :return: A list of :py:class:`mlflow.entities.Metric` entities if logged, else empty list
+        """
+
+        # NB: Pagination for this API is not supported in FileStore or SQLAlchemyStore. The
+        # argument `max_results` is used as a pagination activation flag. If the `max_results`
+        # argument is not provided, this API will return a full metric history event collection
+        # without the paged queries to the backend store.
+        if max_results is None:
+            return self._get_metric_history(run_id, metric_key, None, None)
+        else:
+            return self._get_metric_history(run_id, metric_key, max_results, page_token)
+
     @abstractmethod
-    def get_metric_history(self, run_id, metric_key):
+    def _get_metric_history(self, run_id, metric_key, max_results, page_token):
```

Comment:
I think it's okay for us to include keyword args with default values in the `get_metric_history` method signature. Can we preserve the single-method approach here and decorate `get_metric_history` with `@abstractmethod`?

Reply:
of course. Refactoring now